DogScepterLib/Core/Models/GMObject.cs

```diff
@@ -1,196 +1,362 @@
-            PhysicsKinematic = reader.ReadWideBoolean();
-            PhysicsVertices = new List<PhysicsVertex>(vertexCount);
-            for (int i = vertexCount; i > 0; i--)
-            {
-                PhysicsVertex v = new PhysicsVertex();
-                v.Deserialize(reader);
-                PhysicsVertices.Add(v);
-            }
-            Events = new GMUniquePointerList<GMUniquePointerList<Event>>();
-            Events.Deserialize(reader);
+            X = reader.ReadSingle();
+            Y = reader.ReadSingle();
         }
+    }
 
-        public override string ToString()
+    /// <summary>
+    /// An event that a <see cref="GMObject"/> uses.
+    /// </summary>
+    public class Event : IGMSerializable
+    {
+        /// <summary>
+        /// The subtype of this event.
+        /// </summary>
+        /// <remarks>GameMaker suffixes <see cref="GMCode"/>.<see cref="GMCode.Name"/>, which
+        /// <see cref="Action"/>.<see cref="Action.CodeID"/> references, with this ID.</remarks>
+        //TODO: UMT has this as an UINT, while this here is an INT. Who's correct?
+        public int Subtype;
+
+        /// <summary>
+        /// A list of <see cref="Action"/>s this event executes.
+        /// </summary>
+        /// <remarks>TODO: Seems to always have 1 entry, maybe the games using drag-and-drop code are different. Research if that's really the case.</remarks>
+        public GMUniquePointerList<Action> Actions;
+
+        public void Serialize(GMDataWriter writer)
         {
-            return $"Object: \"{Name.Content}\"";
+            writer.Write(Subtype);
+            Actions.Serialize(writer);
         }
 
-        public class PhysicsVertex : IGMSerializable
+        public void Deserialize(GMDataReader reader)
         {
-            public float X, Y;
-
-            public void Serialize(GMDataWriter writer)
-            {
-                writer.Write(X);
-                writer.Write(Y);
-            }
-
-            public void Deserialize(GMDataReader reader)
-            {
-                X = reader.ReadSingle();
-                Y = reader.ReadSingle();
-            }
+            Subtype = reader.ReadInt32();
+            Actions = new GMUniquePointerList<Action>();
+            Actions.Deserialize(reader);
         }
 
-        public class Event : IGMSerializable
+        /// <summary>
+        /// An action that an <see cref="Event"/> can execute.
+        /// </summary>
+        public class Action : IGMSerializable
         {
-            public int Subtype;
-            public GMUniquePointerList<Action> Actions;
+            //TODO: a lot of unknown hardcoded values, from pre studio DnD era. Nowadays, DnD blocks are compiled to bytecode.
+            // Some of the attribute names may also be wrong, most of them have been taken from here:
+            // https://github.com/WarlockD/GMdsam/blob/26aefe3e90a7a7a1891cb83f468079546f32b4b7/GMdsam/GameMaker/ChunkTypes.cs#L466
```

Comment:
These weren't actually taken from there, these are the real names (you can see them in the 1.x project files)